src/elements/input/input.svelte

```diff
@@ -95,19 +95,49 @@ const handleInput = () => {
     return;
   }
 
-  if (type === 'number' && input.value.endsWith('.')) {
-    return;
+  if (type === 'number') {
+    // ensure input of chars that form valid numbers
+    let numString = '';
+
+    let pointCount = 0;
+    let minusCount = 0;
+    let plusCount = 0;
+    let eCount = 0;
+
+    for (const c of input.value) {
```

Comment:
this is throwing a lint error for cognitive complexity
```Refactor this function to reduce its Cognitive Complexity from 19 to the 18 allowed  sonarjs/cognitive-complexity```
but I'm struggling to find a way to make it simpler... would appreciate any ideas

Reply:
what is the goal of this function? To make sure the string is a valid float or a number? Should maybe ad the goal of the loop here as well as a comment